app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,10 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      order.line_items.each do |line_item|
+        line_item.update_column(:pre_tax_amount, line_item.discounted_amount)
```

Comment:
The avatax gem overwrites the methods that set it [here](https://github.com/solidusio/solidus_avatax/blob/master/app/models/spree/tax_rate_decorator.rb#L11-L19), so I think it needs to happen in this gem.